newIDE/app/src/MainFrame/index.js

```diff
@@ -478,7 +479,9 @@ const MainFrame = (props: Props) => {
           currentFileMetadata: null,
           editorTabs: closeProjectTabs(state.editorTabs, currentProject),
         };
-      }).then(() => {});
+      }).then(() => {
+        updateDiscordRichPresence();
+      });
```

Comment:
Important: let's remove the various calls to updateDiscordRichPresence and instead let's make a single one. But where and when?

There is something that is regularly re-rendered, it's `MainFrame`. It also has access to the `project`.
We could call updateDiscordRichPresence in MainFrame but renders in React should be side-effect free. Instead we can make a custom React hook :)

Add in MainFrame a call to `useDiscordRichPresence(project)` and
`useDiscordRichPresence` is a hook like this:

```
const useDiscordRichPresence = (project: ?Project) => {
  const [lastCallTime, setLastCallTime] = React.useState(0);
  React.useEffect(() => {
    if (performance.now() - lastCallTime > 60000) {
      setLastCallTime(performance.now());
      updateDiscordRichPresence({ project });
    }, [project]);
  });
}
```